lib/dry/schema/dsl.rb

```diff
@@ -176,7 +176,7 @@ def optional(name, &block)
       def key(name, macro:, &block)
         raise ArgumentError, "Key +#{name}+ is not a symbol" unless name.is_a?(::Symbol)
 
-        set_type(name, Types::Any)
+        set_type(name, Types::Any.meta(default: true))
```

Comment:
Does it make more sense to use a constant `DefaultType = Types::Any.dup.freeze` and then compare with `equal?`?

Reply:
@flash-gordon it actually doesn't work in some cases, so I used meta instead, which is carried over in these cases and so this check works.